### PR TITLE
Site: new placeholders for nightly version, tag and unstable suffix

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

File: site/in-dev/gc.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ The Nessie GC tool can be downloaded from the [GitHub
 Releases](https://github.com/projectnessie/nessie/releases) page, for example:
 
 ```shell
-curl -L -o nessie-gc.jar https://github.com/projectnessie/nessie/releases/download/nessie-::NESSIE_VERSION::/nessie-gc-::NESSIE_VERSION::.jar
+curl -L -o nessie-gc.jar https://github.com/projectnessie/nessie/releases/download/nessie-::NESSIE_VERSION_BASE::/nessie-gc-::NESSIE_VERSION_BASE::.jar
```
Comment (adutra) [inline site/in-dev/gc.md:36]: Obviously, this link will give 404. I don't think there is much we can do about it.

Comment (snazy) [reply site/in-dev/gc.md:36]: I think it's better to just link the GC options on the downloads page and remove the curl example

File: site/bin/_lib.sh [hunk 2]
```diff
@@ -314,8 +324,11 @@ release() {
   echo "     ... replace title in versioned mkdocs.yml"
   sed -i'' -E "s/(^site_name:[[:space:]]+).*$/\1\"Nessie ${RELEASE_VERSION}\"/" "${target}/mkdocs.yml"
 
-  echo "     ... replace version placeholder in versioned docs"
-  find "${target}" -name "*.md" -exec sed -i "s/::NESSIE_VERSION::/${RELEASE_VERSION}/g" {} \;
+  echo "     ... replace version placeholders in versioned docs"
+  find "${target}" -name "*.md" -exec sed -i='' "s/::NESSIE_VERSION::/${RELEASE_VERSION}/g" {} \;
```
Comment (adutra) [inline site/bin/_lib.sh:328]: According to this [SO question](https://stackoverflow.com/questions/12272065/sed-undefined-label-on-macos), the syntax `-i=''` should be portable across different OSes (just `-i` gives an error on macOS).

File: site/in-dev/gc.md [hunk 3]
```diff
@@ -142,7 +136,7 @@ spec:
     spec:
       containers:
       - name: nessie-gc
-        image: ghcr.io/projectnessie/nessie-gc:::NESSIE_VERSION::
+        image: ghcr.io/projectnessie/nessie-gc::NESSIE_UNSTABLE:::::NESSIE_TAG::
```
Comment (snazy) [inline site/in-dev/gc.md:139]:
```suggestion
        image: ghcr.io/projectnessie/nessie-gc::NESSIE_DOCKER_SUFFIX::
```

File: site/bin/_lib.sh (hunk 2 above)
Comment (snazy) [reply site/bin/_lib.sh:328]: Seems to work, yes

File: site/bin/_lib.sh [hunk 4]
```diff
@@ -108,6 +112,12 @@ create_nightly () {
   mv build/versions/nightly/docs/mkdocs.yml build/versions/nightly
   rm build/versions/nightly/docs/index-release.md
 
+  echo "     ... replace version placeholders in versioned docs"
+  find build/versions/nightly/docs -name "*.md" -exec sed -i='' "s/::NESSIE_VERSION::/${version}/g" {} \;
+  find build/versions/nightly/docs -name "*.md" -exec sed -i='' "s/::NESSIE_VERSION_BASE::/${version_base}/g" {} \;
```
[the remark this answers is not in the document]
Comment (adutra) [reply site/bin/_lib.sh]: Removed.

**Review by snazy — APPROVED** [review]

**Pull request merged**